smartsim/_core/utils/telemetry/telemetry.py

```diff
@@ -316,12 +318,24 @@ async def on_timestep(self, timestamp: int) -> None:
         # consider not using name to avoid collisions
         m_jobs = [job for job in self._tracked_jobs.values() if job.is_managed]
         if names := {entity.name: entity for entity in m_jobs}:
-            step_updates = self._launcher.get_step_update(list(names.keys()))
-
-            for step_name, step_info in step_updates:
-                if step_info and step_info.status in TERMINAL_STATUSES:
-                    completed_entity = names[step_name]
-                    await self._to_completed(timestamp, completed_entity, step_info)
+            step_updates: t.List[t.Tuple[str, t.Optional[StepInfo]]] = []
+
+            try:
+                task_names = list(names.keys())
+                updates = self._launcher.get_step_update(task_names)
+                step_updates.extend(updates)
+                logger.debug(f"Retrieved updates for: {task_names}")
+            except Exception:
+                logger.warning(f"Telemetry step updates failed for {names.keys()}")
+
+            try:
+                for step_name, step_info in step_updates:
+                    if step_info and step_info.status in TERMINAL_STATUSES:
+                        completed_entity = names[step_name]
+                        await self._to_completed(timestamp, completed_entity, step_info)
+            except Exception as ex:
+                msg = f"An error occurred getting step updates on {names}"
+                logger.error(msg, exc_info=ex)
```

Comment:
This PR seems to have grown in scope some what. Are these telemetry monitor changes strictly related socket/authenticator changes? Is there any chance we could separate them into two or more PRs in case we need to roll one back?

If not consider updating the PR title/descriptions to encompass more of these TM related changes.

Reply:
Split in two